flake.nix

```diff
@@ -9,13 +9,29 @@
   };
 
   outputs = { self, nixpkgs, flake-utils, ... }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        pkgs = nixpkgs.legacyPackages.${system};
-        taxi = (import ./pkgs.nix { inherit pkgs; });
-      in {
-        packages = { inherit taxi; };
-        defaultPackage = taxi;
-        devShell = import ./shell.nix { inherit pkgs; };
-      });
+    let
+      # Recursively merge a list of attribute sets. Following elements take
+      # precedence over previous elements if they have conflicting keys.
+      recursiveMerge = with nixpkgs.lib; foldl recursiveUpdate { };
+    in
+    recursiveMerge [
+      (flake-utils.lib.eachDefaultSystem
+        (system:
+          let
+            pkgs = nixpkgs.legacyPackages.${system};
+            taxi = (import ./pkgs.nix { inherit pkgs; });
+          in
+          {
+            packages = { inherit taxi; };
+            defaultPackage = taxi;
+            devShell = import ./shell.nix { inherit pkgs; };
+          })
+      )
+      {
+        overlay = final: prev: rec {
+          # Prevents collision with taxi from nixpkgs (sftp app)
+          taxi-cli = self.packages.${prev.stdenv.hostPlatform.system}.taxi;
```

Comment:
I think we should avoid pointing base Nix packages to other packages as this could lead to unpleasant surprises (eg. if a Nix package depends on `taxi`).

I’m not sure I understand what problem this change solves: you can already use this Taxi package on your system without any conflict, eg. by importing the Taxi flake and then using it as an overlay on your nixpkgs:

```nix
{
  inputs.taxi = "github:sephii/taxi";
  outputs = { self, nixpkgs, taxi }:
    let
      pkgs = import nixpkgs { overlays = [ (final: prev: { taxi-timesheets = taxi.packages.x86_64-linux.taxi; }) ]; };
    in
    ...
}
```

Reply:
The comment was more geared to calling it `taxi-cli` rather than calling it straight `taxi`.

The big advantage of exposing an overlay output to the flake is that it can be consumed by other flakes directly. If you take a look at my current NixOS configuration here: https://github.com/jocelynthode/nixos-config/blob/master/flake.nix#L39 you can see that I use it this way without having to write the overlay myself.

Reply:
Ah yes sorry, I didn’t notice `taxi-cli` refers to `self.packages`.